scripts/high_level_motion_unit_test.py

```diff
@@ -0,0 +1,173 @@
+    client.send_goal(goal)
+    # Waits for the server to finish performing the action.
+    client.wait_for_result()
+    assert(client.get_result().success == False)
+
+    rospy.sleep(2.0)
+
+    rospy.logwarn("RESET STATE")
+
+    goal = QuadMoveGoal(movement_type="takeoff")
+    # Sends the goal to the action server.
+    client.send_goal(goal)
+    # Waits for the server to finish performing the action.
+    client.wait_for_result()
+    rospy.logwarn("Takeoff success: {}".format(client.get_result()))
+
+    rospy.sleep(2.0)
+
+    rospy.logwarn("Preparing to test illegal state #4")
+
+    goal = QuadMoveGoal(movement_type="hit_roomba", frame_id = roomba_id)
+    # Sends the goal to the action server.
+    client.send_goal(goal)
+    # Waits for the server to finish performing the action.
+    client.wait_for_result()
+    rospy.logwarn("Hit Roomba success: {}".format(client.get_result()))
+
+    rospy.sleep(2.0)
+
+    rospy.logwarn("Testing illegal state #4")
+
+    goal = QuadMoveGoal(movement_type="track_roomba", frame_id=roomba_id, 
+        time_to_track=5.0, x_overshoot=.5, y_overshoot=.5)
+    # Sends the goal to the action server.
+    client.send_goal(goal)
+    # Waits for the server to finish performing the action.
+    client.wait_for_result()
+    assert(client.get_result().success == False)
+    
+    rospy.sleep(2.0)
+    
+    rospy.logwarn("RESET STATE")
+
+    goal = QuadMoveGoal(movement_type="height_recovery")
+    # Sends the goal to the action server.
+    client.send_goal(goal)
+    # Waits for the server to finish performing the action.
+    client.wait_for_result()
+    rospy.logwarn("Height Recovery success: {}".format(client.get_result()))
+
+    rospy.sleep(2.0)
+
+    rospy.logwarn("Sending Test Task")
+
+    goal = QuadMoveGoal(movement_type="test_task", takeoff_height=0.75)
+    # Sends the goal to the action server.
+    client.send_goal(goal)
+    # Waits for the server to finish performing the action.
+    client.wait_for_result()
+    rospy.logwarn("Test Task success: {}".format(client.get_result()))
+
+    rospy.sleep(2.0)
+
+    rospy.logwarn("RESET STATE")
+    goal = QuadMoveGoal(movement_type="land")
+    # Sends the goal to the action server.
+    client.send_goal(goal)
+    # Waits for the server to finish performing the action.
+    client.wait_for_result()
+    rospy.logwarn("Land success: {}".format(client.get_result()))
+
+    rospy.logwarn("Testing is complete")
+    
+def _receive_roomba_status(data):
+    global roomba_array
+    roomba_array = data
+
+if __name__ == '__main__':
+    try:
+        roomba_array = []
+        # Initializes a rospy node so that the SimpleActionClient can
+        # publish and subscribe over ROS.
+        _roomba_status_sub = rospy.Subscriber('roombas', 
+                         OdometryArray, _receive_roomba_status)
+
+        rospy.init_node('high_level_motion_unit_test')
+        test()
+        while not rospy.is_shutdown():
```

Comment:
Just call `rospy.spin` here instead of this loop

Reply:
fixed